### PR TITLE
Feat: add secretes

### PR DESCRIPTION
**Pull request by Trishisingh**
Added  approach to add secrets to Azure KV

### CODE REVIEW

File: build/azDevOps/azure/air-infrastructure-data-vars.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ variables:
   - name: tf_state_container
     value: tfstate
   - name: tf_state_key_dev
-    value: $(domain)-dev
+    value: $(domain)_dev
```
Comment (adurkan-amido) [inline build/azDevOps/azure/air-infrastructure-data-vars.yml:37]: Any reason it's `_dev` and `-prod`

File: deploy/azure/infra/vars.tf [hunk 2]
```diff
@@ -144,3 +144,17 @@ variable "container_access_type" {
...
+
+variable "kv_secrets" {
+  type        = list(string)
+  description = "Specifies the name of the Key Vault Secrets. The secrets' values will need to be updated directly once deployed. Existing secrets with the same name will not be overwritten."
+  default     = ["secret1", "secret2", "secret3"]
+}
+
+
+variable "contributor_object_ids" {
+  description = "A list of Azure active directory user,group or application object ID's that will have contributor role to the key vault"
```
Comment (adurkan-amido) [inline deploy/azure/infra/vars.tf:157]:
```suggestion
  description = "A list of Azure Active Directory user, group or application object IDs that will have contributor role for  the Key Vault."
```

File: deploy/azure/infra/main.tf [hunk 3]
```diff
@@ -143,3 +143,18 @@ resource "azurerm_data_factory_linked_service_data_lake_storage_gen2" "linked_ad
   use_managed_identity = true
   url                  = module.adls_default.primary_dfs_endpoints[1]
 }
+
+# ADD secrets to KV, please note this is just going to add secret name to KV the actual value of that secret needs to be updated manually in Azure Key Valut. Existing secrets with the same name will not be overwritten.
```
Comment (adurkan-amido) [inline deploy/azure/infra/main.tf:147]:
```suggestion
# Add secrets to KV. Please note this is just going to add secret names to KV. The actual value of that secret needs to be updated manually in Azure Key Vault. Existing secrets with the same name will not be overwritten.
```

**Review by adurkan-amido — COMMENTED** [review]
Looks good apart from small comments.

File: build/azDevOps/azure/air-infrastructure-data-vars.yml (hunk 1 above)
Comment (Trishisingh) [reply build/azDevOps/azure/air-infrastructure-data-vars.yml:37]: updated now, was doing some local testing

**Review by adurkan-amido — APPROVED** [review]

**Pull request merged**